### PR TITLE
Implement `group by` and friends

### PR DESCRIPTION
**Pull request by mcls**
Trying to get my head around how this would work with ROM.

I wrote an integration spec to show what kind of things I'm trying to do.

What do you think?

### CODE REVIEW

**Comment by solnic** [comment]
:+1: could you add yard docs and push this to master? :)

File: lib/rom/sql/relation.rb [hunk 1]
```diff
@@ -39,6 +39,26 @@ def project(*names)
...
+
+      # @api public
+      def select_append(*args, &block)
+        __new__(dataset.__send__(__method__, *args, &block))
+      end
+
+      # @api public
+      def select_group(*args, &block)
+        __new__(dataset.__send__(__method__, *args, &block))
+      end
```
Comment (gotar) [inline lib/rom/sql/relation.rb]: why not to use some alias method? all those methods do the same for me

Comment (mcls) [inline lib/rom/sql/relation.rb]: They're being defined separately so they can be documented. But I guess the `__new__(dataset.__send__(__method__, *args, &block))` lines "DRY'ed" a bit.

Comment (solnic) [inline lib/rom/sql/relation.rb]: we could dry it up but I'm not sure if it's worth it, those methods will not change at all. As @mcls pointed out it's just to define the interface explicitly and document it.

**Pull request merged**